src/main/java/pl/beata/zajecia22_zad1/UserController.java

```diff
@@ -0,0 +1,97 @@
+        if (name.equals("")) {
+            return "redirect:/err.html";
+        } else {
+            User newUser = new User(name, surname, age);
+            userRepository.add(newUser);
+            return "redirect:/success.html";
+        }
+    }
+
+
+    @RequestMapping("/add")
+    public String addUser(@RequestParam(name = "imie") String name, @RequestParam(name = "nazwisko") String surname, @RequestParam(name = "wiek") Integer age) {
+        if (name.equals("")) {
+            return "err.html";
+        } else {
+            User newUser = new User(name, surname, age);
+            userRepository.add(newUser);
+            return "success.html";
+        }
+    }
+
+    @ResponseBody
+    @RequestMapping("/users")
+    public String getUsers() {
+        List<User> usersList = userRepository.getUsers();
+        String result = "";
+        for (User user : usersList) {
+            result += user.getName() + " " + user.getSurname() + " wiek: " + user.getAge() + "<br/>";
+        }
+        return result;
+    }
+
+    @ResponseBody
+    @RequestMapping("/filterUsers")
+    public String filterUsers(@RequestParam(name = "imie") String name, @RequestParam(name = "nazwisko", defaultValue = "xxx", required = false) String surname, @RequestParam(name = "wiek", defaultValue = "-1", required = false) Integer age) {
+        List<User> usersList = getUsers(name, surname, age);
+
+        String result = "";
+        for (User user : usersList) {
+            result += user.getName() + " " + user.getSurname() + " wiek: " + user.getAge() + "<br/>";
+        }
+        return result;
+    }
+
+    @ResponseBody
+    @RequestMapping("/deleteUsers")
+    public String deleteUsers(@RequestParam(name = "imie") String name, @RequestParam(name = "nazwisko", defaultValue = "xxx", required = false) String surname, @RequestParam(name = "wiek", defaultValue = "-1", required = false) Integer age) {
+        List<User> usersToBeRemoved = getUsers(name, surname, age);
+        userRepository.getUsers().removeAll(usersToBeRemoved);
+
+        String result = "";
+        for (User user : userRepository.getUsers()) {
+            result += user.getName() + " " + user.getSurname() + " wiek: " + user.getAge() + "<br/>";
+        }
+        return result;
+    }
+
+    private List<User> getUsers(@RequestParam(name = "imie") String name, @RequestParam(name = "nazwisko", defaultValue = "xxx", required = false) String surname, @RequestParam(name = "wiek", defaultValue = "-1", required = false) Integer age) {
+        List<User> usersList = userRepository.getUsers();
+
+        if (!name.equals("")) {
+            usersList = UserUtils.filterUsers(usersList, user -> user.getName().equalsIgnoreCase(name));
+        }
+
+        if (!surname.equals("xxx")) {
```

Comment:
Zapinanie się na "magiczne" liczby oraz słowa to nie jest dobry sposób. Jedyne co Cie tutaj powinno interesować to to czy coś jest null czy ew pustym stringiem.